internal/pkg/agent/application/upgrade/step_download_test.go

```diff
@@ -209,5 +257,43 @@ func TestDownloadWithRetries(t *testing.T) {
 			require.Equal(t, fmt.Sprintf("download attempt %d", i+1), logs[(2*i)].Message)
 			require.Contains(t, logs[(2*i+1)].Message, "unable to download package: download failed; retrying")
 		}
+
+		// Check that upgradeDetails.Metadata.RetryUntil was set at some point
+		// during the retryable download and then check that it was never unset,
+		// since we didn't have a successful download.
+		require.GreaterOrEqual(t, *upgradeDetailsRetryUntil, minRetryDeadline)
+		require.False(t, *upgradeDetailsRetryUntilWasUnset)
+		require.Equal(t, *upgradeDetailsRetryUntil, *upgradeDetails.Metadata.RetryUntil)
+
+		// Check that upgradeDetails.Metadata.RetryableErrorMsg was set at some point
+		// during the retryable download and then check that it was never unset,
+		//since we didn't have a successful download.
+		require.NotEmpty(t, *upgradeDetailsRetryableErrorMsg)
+		require.Equal(t, *upgradeDetailsRetryableErrorMsg, upgradeDetails.Metadata.RetryableErrorMsg)
 	})
 }
+
+func mockUpgradeDetails(parsedVersion *agtversion.ParsedSemVer) (*details.Details, *time.Time, *bool, *string) {
```

Comment:
There is a lot happening and being returned here, thus some doc comment would be good

Reply:
Added in bc950ce9bc.